source/shadow-tasks/shadow_device_task.c

```diff
@@ -0,0 +1,816 @@
+    "\"powerOn\":%1u"                     \
+    "}"                                   \
+    "},"                                  \
+    "\"clientToken\":\"%06lu\""           \
+    "}"
+
+/**
+ * @brief The expected size of #SHADOW_REPORTED_JSON.
+ *
+ * Since all of the format specifiers in #SHADOW_REPORTED_JSON include a length,
+ * its actual size can be pre-calculated at compile time from the difference between
+ * the lengths of the format strings and their formatted output. We must subtract 2
+ * from the length as according the following formula:
+ * 1. The length of the format string "%1u" is 3.
+ * 2. The length of the format string "%06lu" is 5.
+ * 3. The formatted length in case 1. is 1 ( for the state of powerOn ).
+ * 4. The formatted length in case 2. is 6 ( for the clientToken length ).
+ * 5. Thus the additional size of our format is 2 = 3 + 5 - 1 - 6 + 1 (termination character).
+ *
+ * Custom applications may calculate the length of the JSON document with the same method.
+ */
+#define shadowexampleSHADOW_REPORTED_JSON_LENGTH       ( sizeof( shadowexampleSHADOW_REPORTED_JSON ) - 2 )
+
+/**
+ * @brief Time in ms to wait between checking for updates to report.
+ */
+#define shadowexampleMS_BETWEEN_REPORTS                ( 15000U )
+
+/**
+ * @brief This demo uses task notifications to signal tasks from MQTT callback
+ * functions. shadowexampleMS_TO_WAIT_FOR_NOTIFICATION defines the time, in ticks,
+ * to wait for such a callback.
+ */
+#define shadowexampleMS_TO_WAIT_FOR_NOTIFICATION       ( 5000 )
+
+/**
+ * @brief The maximum amount of time in milliseconds to wait for the commands
+ * to be posted to the MQTT agent should the MQTT agent's command queue be full.
+ * Tasks wait in the Blocked state, so don't use any CPU time.
+ */
+#define shadowexampleMAX_COMMAND_SEND_BLOCK_TIME_MS    ( 200 )
+
+/**
+ * @brief Defines the structure to use as the command callback context in this
+ * demo.
+ */
+struct CommandContext
+{
+    bool xReturnStatus;
+};
+
+extern MQTTAgentContext_t xGlobalMqttAgentContext;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The simulated device current power on state.
+ */
+static uint32_t ulCurrentPowerOnState = 0;
+
+/**
+ * @brief The last reported state. It is initialized to and invalid value so that
+ * an update is initially sent.
+ */
+static uint32_t ulReportedPowerOnState = 2;
+
+/**
+ * @brief When we send an update to the device shadow, and if we care about
+ * the response from cloud (accepted/rejected), remember the clientToken and
+ * use it to match with the response. We set this to 0 when not waiting on a
+ * response.
+ */
+static uint32_t ulClientToken = 0U;
+
+/**
+ * @brief The handle of this task. It is used by callbacks to notify this task.
+ */
+static TaskHandle_t xShadowDeviceTaskHandle;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Subscribe to the used device shadow topics.
```

Comment:
Yeah, I agree we should follow up and have a common subscription callback in another PR. 
That is meant to be used. Since not all shadow topics are used.